grr/server/grr_response_server/fleetspeak_utils.py

```diff
@@ -129,6 +130,6 @@ def FetchClientResourceUsageRecordsFromFleetspeak(
   res = fleetspeak_connector.CONN.outgoing.FetchClientResourceUsageRecords(
       admin_pb2.FetchClientResourceUsageRecordsRequest(
           client_id=GRRIDToFleetspeakID(client_id), limit=limit))
-  if not res or not res.records or not res.records[0]:
+  if not res.records:
```

Comment:
Yes, `res.records` is a repeated field.

I thought that this can solve a problem where the response returned from the gRPC service is not what we expect; from instance, [an error that rises](https://github.com/google/fleetspeak/blob/616a079d4c55e4637574ddda1804ce36d68cde18/fleetspeak/src/server/admin/admin.go#L253) from Fleetspeak's side. So if we get an [`idErr`](https://github.com/google/fleetspeak/blob/298e9ae7573cf0c2f29b0c1e68a0a11370259c53/fleetspeak/src/common/ids.go#L41), then `FetchClientResourceUsageRecords` on GRR's side will essentially not crash.

I tried researching it a little bit, but at the moment did not arrive at a final conclusion; is it possible that such an error within the gRPC service can crash the HTTP server? That's a situation we want to avoid in a case where a Grafana user accidentally enters a wrong client ID. My intention was that checking whether `res.records` is falsey will imply that it is either empty (in this case it does not matter) or it does not exist as a property of the `res`, in which case, it isn't a valid instance of `FetchClientResourceUsageRecordsResponse`.

I agree with you that in a case where it is not possible, then this if statement is unnecessary.
What do you think?

Reply:
Protobuf are strictly typed - so even if the RPC fails, if the response parses as your expected Protobuf message, you can expect `res.records` to be a list (possibly empty).

If the RPC fails, `FetchClientResourceUsageRecords` should fail as well - it's only a wrapper around the RPC. There is no benefit in returning an empty list if the underlying call failed due to some error. This can actually make debugging issues very hard, because underlying issues are concealed.